torch/_jit_internal.py

```diff
@@ -26,17 +26,81 @@ def createResolutionCallbackFromEnv(lookup_base):
     You should not use this directly, it should only be used from the other
     createResolutionCallbackFrom* functions.
     """
+    def try_build_type(base_mod, subexp_mod):
```

Comment:
i started having logic in multiple places and it was still failing in cases like a unit test that tries Tensor[Tensor] and expects a particular exception.